services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -4801,6 +4801,23 @@ public Map<String, StoreDataAudit> getClusterStaleStores(String clusterName) {
     return retMap;
   }
 
+  /**
+   * This function iterates over all of Helix Parent Admin's child controllers,
+   * in order to obtain the stores and respective store information of each child controller,
+   * and then filter out the stores that are ready for compaction.
+   * @param clusterName, the name of the cluster to search for stores that are ready for compaction
+   * @return a list of StoreInfo objects of stores that are ready for compaction
+   */
+  @Override
+  public ArrayList<StoreInfo> getStoresForCompaction(String clusterName) {
+    throw new UnsupportedOperationException("This function is implemented in VeniceHelixAdmin.");
+  }
+
+  @Override
+  public void compactStore(String storeName) {
+    // TODO
```

Comment:
Can I confirm that `VPHA::compactStore()` does nothing since the compaction is done by child controllers?

Can I also confirm that `VHA::compactStore()` only calls `CompactionManager::compactStore()` and let `CompactionManager` handle the trigger response & logging?

Reply:
Depends on what sort of thing happens within `CompactionManager::compactStore`. E.g., if you are invoking other APIs to get some other store metadata. My guess is it should happen on the child controller and also the leader child controller (potentially) to make sure it sees the most up-to-date store metadata e.g., version information (if at all it needs to invalidate some of the request parameters).

Reply:
Let's have `CompactionManager::compactStore` log the response status but have `VHA::compactStore` receive and handle errors. 

We can think of `CompactionManager` as an "irresponsible" class with low visibility, and all it does is execute and hands off any results back upstream.